### ISSUE TITLE
[Feature Request] Have Characters/Spells Link in a Dynamic Way

### ISSUE THREAD

**Issue by ZachMyers3** [ZachMyers3/flask-dnd#14]
For a given character, there's only a number of spells available depending on various attributes a character has (Race/Classes/Level/etc.). It should be possible to pull this information and run the spell list through these filters to generate a list of these spells for the character.

Expanding on this, there will need to be a way for this list to dynamically regenerate when any of these attributes change, and a way for some of the spells to be selected as active/in-use for the given character (for a future usage as a spell sheet for example).

```
example generated spell list for a character:
{
  "name": "Test Guy",
  ...
  "spells": [
    {
      "spell_id": "abcdefg",   // link to the id from the spells collection
      "in_use": true           // boolean to show that the character is using this available spell
    },
    {
      "spell_id": "hijklmno",
      "in_use": false
    }
  ]
}
```

**Comment by ZachMyers3** [comment]
Current the calc_spells function is not pulling spells that have subclasses. Maybe the query for a spell list isn't working properly.

**Comment by ZachMyers3** [comment]
Feature complete, bug fixed in e0aaeb1272779c77cef2060c742ed174061555b1.

The query to gather spells was explicitly looking for spells with ```classes``` object with the given characters subclass like so:

```python
class_spells = list(mongo.db.spells.find({
            "classes": {
                "class": cclass["name"]
            }
        }))
```

This didn't allow for any additional attributes to be within the classes object. This was fixed by adjusting the query as shown:

```python
class_spells = list(mongo.db.spells.find({
            "classes.class": cclass["name"]
        }))
```

The change allows for additional information to be present within the object for the given query, and is only caring about the classes class name string, which is what is intended.
